src/amplitude-client.js

```diff
@@ -556,32 +556,36 @@ AmplitudeClient.prototype._unsentCount = function _unsentCount() {
 };
 
 /**
- * Send events if ready. Returns true if events are sent.
- * @private
+ * Flushes batched events if ready. Is periodically called by SDK,
+ * but can be manually called by client. Behavior is dependent on configuration
+ * If `batchEvents === false`, it is called after `logEvent`.
+ * If `batchEvents === true`, then events are sent only when batch criterias are met.
+ * @public
+ * @returns {boolean} true if events are sent. false if they aren't or are scheduled for a later time
  */
-AmplitudeClient.prototype._sendEventsIfReady = function _sendEventsIfReady() {
+AmplitudeClient.prototype.flushEvents = function flushEvents() {
   if (this._unsentCount() === 0) {
     return false;
   }
 
   // if batching disabled, send any unsent events immediately
   if (!this.options.batchEvents) {
-    this.sendEvents();
+    this._sendEvents();
     return true;
   }
 
   // if batching enabled, check if min threshold met for batch size
   if (this._unsentCount() >= this.options.eventUploadThreshold) {
-    this.sendEvents();
+    this._sendEvents();
     return true;
   }
 
-  // otherwise schedule an upload after 30s
+  // otherwise schedule an upload after `eventUploadPeriodMillis`
   if (!this._updateScheduled) {  // make sure we only schedule 1 upload
     this._updateScheduled = true;
     setTimeout(function() {
```

Comment:
nit: curious, couldn't we change this from `function() {...}.bind(this)` to `() => { ... }`

Reply:
added